core/ui.py

```diff
@@ -12,29 +19,24 @@ def add_row_number(self, row_int, total_rows):
         else:
             return str(row_int + 1) + ' '
     
-    def add_shot_marker(self, board_state, row, column, ship_symbols):
-        if board_state[row][column] is None or board_state[row][column] in ship_symbols:
+    def add_shot_marker(self, board_state, row, column, all_ships):
```

Comment:
Why aren't row and column combined into something? Does it seem weird for add shot marker to need to know about ships?